pkgs/by-name/ov/overskride/package.nix

```diff
@@ -0,0 +1,57 @@
+{ lib, fetchFromGitHub, stdenv, rustPlatform, cargo, rustc, meson, ninja
+, pkg-config, wrapGAppsHook4, desktop-file-utils, appstream-glib
+, blueprint-compiler, dbus, gtk4, libadwaita, bluez, libpulseaudio }:
+let
+
+  pname = "overskride";
+  version = "0.5.6";
+  owner = "kaii-lb";
+
+  src = fetchFromGitHub {
+    inherit owner;
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-syQzHHT0s15oj8Yl2vhgyXlPI8UxOqIXGDqFeUc/dJQ=";
+  };
+
+in stdenv.mkDerivation {
+
+  inherit src pname version;
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    hash = "sha256-NEsqVfKZqXSLieRO0BvQGdggmXXYO15qVhbfgAFATPc=";
+  };
+
+  nativeBuildInputs = [
+    rustPlatform.cargoSetupHook
+    pkg-config
+    wrapGAppsHook4
+    desktop-file-utils
+    appstream-glib
+    blueprint-compiler
+    meson
+    ninja
+    cargo
+    rustc
+  ];
+
+  buildInputs = [ dbus gtk4 libadwaita bluez libpulseaudio ];
+
+  preFixup = ''
+    glib-compile-schemas $out/share/gsettings-schemas/${pname}-${version}/glib-2.0/schemas
+  '';
+
+  meta = with lib; {
+    description =
+      "A Bluetooth and Obex client that is straight to the point, DE/WM agnostic, and beautiful";
+    homepage = "https://github.com/${owner}/${pname}";
+    changelog = "https://github.com/${owner}/${pname}/blob/v${version}/CHANGELOG.md";
+    license = licenses.gpl3Only;
+    mainProgram = pname;
```

Comment:
I wouldn't say that `finalAttrs.pname` and `mainProgram` are necessarily semantically different?

Reply:
they absolutely are, otherwise what's the whole point of requiring this to be set manually

Reply:
> otherwise what's the whole point of requiring this to be set manually

For the sake of discussion: Because it is in principle possible for them to be different, and setting one of them automatically would be wrong in those cases.

Reply:
exactly. so that's why they mean different things.

Reply:
> If someone overrides this with a different pname, this will actually break. Please correct this.

Thanks. All I needed was a good reason to know this is not just a nitpick 😃